docs/dev/lwt-implementation.md

```diff
@@ -0,0 +1,166 @@
+            return True  # Accepted
+        return False  # Rejected
+
+class Proposer:
+    def __init__(self, proposer_id: int, acceptors: List[Acceptor]):
+        self.proposer_id = proposer_id
+        self.acceptors = acceptors
+        self.proposal_number = proposer_id  # Start with unique ID-based number
+        self.quorum = len(acceptors) // 2 + 1  # Majority
+
+    def prepare(self, value: str) -> bool:
+        self.proposal_number += 10  # Increment to ensure uniqueness (simplified)
+        promises = 0
+        highest_accepted = None
+
+        for acceptor in self.acceptors:
+            promise, accepted_proposal = acceptor.on_prepare(self.proposal_number)
+            if promise:
+                promises += 1
+                if accepted_proposal and (highest_accepted is None or 
+                                         accepted_proposal.number > highest_accepted.number):
+                    highest_accepted = accepted_proposal
+
+        if promises >= self.quorum:
+            # If we have a majority, pick value: use highest accepted or our own
+            self.value = highest_accepted.value if highest_accepted else value
+            return True
+        return False
+
+    def accept(self) -> bool:
+        """Phase 2: Send accept requests with chosen value."""
+        accepts = 0
+        proposal = Proposal(self.proposal_number, self.value)
+
+        for acceptor in self.acceptors:
+            if acceptor.on_accept(proposal):
+                accepts += 1
+
+        return accepts >= self.quorum  # Consensus if majority accepts
+
+class Learner:
+    def __init__(self):
+        self.learned_value = None
+
+    def learn(self, value: str):
+        self.learned_value = value
+        print(f"Learned consensus value: {value}")
+
+def run_paxos():
+    acceptors = [Acceptor() for _ in range(5)]  # 5 acceptors
+    proposer = Proposer(proposer_id=1, acceptors=acceptors)
+    learner = Learner()
+
+    # Proposer tries to propose a value
+    if proposer.prepare("ValueA"):
+        if proposer.accept():
+            learner.learn(proposer.value)
+
+if __name__ == "__main__":
+    run_paxos()
+```
+
+## Scylla Paxos
+
+Scylla Paxos implementation is more elaborate than the base scheme outlined above, and its inner logic is quite different.
+
+The system.paxos table, besides the standard last promised ballot and last accepted proposal, also stores a most_recent_commit field. This field is used as an opmitization. The learn phase stores the committed value in both the base table and in this field. When the next prepare for this key comes to the node later, this committed value is returned. This allows the coordinator to avoid a separate quorum read in storage_service::cas, provided that the prune process has not yet removed this field. The prune is started when the learn is confirmed from the quorum of nodes. This means that this optimization doesn't apply if LWT requests are rare.
+
```

Comment:
most_recent_commit is not only optimization. It is used to avoid real bugs like this one https://issues.apache.org/jira/browse/CASSANDRA-5062?focusedCommentId=13619810&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13619810

It is needed because in our paxos we reset a state after lear and re-use the same paxos register again.